at_client/lib/src/service/sync_service_impl.dart

```diff
@@ -175,12 +175,18 @@ class SyncServiceImpl implements SyncService, AtSignChangeListener {
       }
 
       _syncInProgress = true;
-      final serverCommitId = await _getServerCommitId();
-      final localCommitId = await _getLocalCommitId();
+      int serverCommitId = await _getServerCommitId();
+      final localCommitIdBeforeSync = await _getLocalCommitId();
       final syncResult = await _sync(serverCommitId, syncRequest);
       _syncComplete(syncRequest);
       syncProgress.syncStatus = syncResult.syncStatus;
-      _informSyncProgress(syncProgress, localCommitId: localCommitId);
+      syncProgress.keyInfoList = syncResult.keyInfoList;
+      serverCommitId = await _getServerCommitId();
```

Comment:
Do we require both calls at line 178 and this?

Reply:
yes.when local is ahead...server commit id will be incremented during sync